File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = {7,7};
        
        for (int i =0; i<lottos.length; ++i){
            for (int j=0; j<win_nums.length; ++j){
                if (lottos[i] == win_nums[j]){
                    answer[0] -= 1;
                    answer[1] -= 1;
                } 
            }
            if (lottos[i] == 0){
                    answer[0] -= 1;
                }
        }
        if(answer[1] == 7) {
            answer[1] = 6;
        }
        if(answer[0] == 7) {
            answer[0] = 6;
        }
            return answer;
    }
}

File: JavaScript/프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    let matches = 0;
    let unknowns = 0;

    for (let num of lottos) {
        if (num === 0) {
            unknowns++;
        } else if (win_nums.includes(num)) {
            matches++;
        }
    }
    const maxRank = Math.min(7 - (matches + unknowns), 6);
    const minRank = Math.min(7 - matches, 6);
    
    return [maxRank, minRank];
}